File: Programmers/pro_12924_숫자의표현.py
def solution(n):
    answer,acc=0,0
    s,e=0,0    
    while(s<n):
        if(acc>n):
            acc-=s
            s+=1
        else:
            if(e<n):
                e+=1
                acc+=e
        if(acc==n):
            answer+=1
    return answer

File: Programmers/pro_12924_숫자의 표현.js
function solution(n) {
    let [lo, hi] = [1, 1]
    let answer = 0;
    let sum = 0
    while (1) {
        if (sum >= n) sum -= lo++;
        else if (hi === n + 1) break;
        else sum += hi++;
        if (sum === n) answer++;
    }
    return answer;
}